### PR TITLE
ENG-12179: Add classification rule field to data label resource

### PR DESCRIPTION
**Pull request by VictorGFM**
## Description of the change

This PR contains the following changes:
- Adds `classification_rule` field to data label resource
- Deprecates `cyral_sidecar_cft_template` data source since it is no longer valid for new sidecars

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Jira issue referenced in commit message and/or PR title

### Testing

Automated tests and manual E2E tests. See automated tests result below:
```log
go test github.com/cyralinc/terraform-provider-cyral/... -v -race -timeout 20m
?       github.com/cyralinc/terraform-provider-cyral    [no test files]
=== RUN   TestNewClient_WhenTLSSkipVerifyIsEnabled_ThenInsecureSkipVerifyIsTrue
2023/08/08 17:18:42 [DEBUG] Init NewClient
2023/08/08 17:18:42 [DEBUG] TokenSource: &{0xc0000a6d98 {0 0} <nil>}
2023/08/08 17:18:42 [DEBUG] End NewClient
--- PASS: TestNewClient_WhenTLSSkipVerifyIsEnabled_ThenInsecureSkipVerifyIsTrue (0.00s)
=== RUN   TestNewClient_WhenTLSSkipVerifyIsDisabled_ThenInsecureSkipVerifyIsFalse
2023/08/08 17:18:42 [DEBUG] Init NewClient
2023/08/08 17:18:42 [DEBUG] TokenSource: &{0xc0000a6de0 {0 0} <nil>}
2023/08/08 17:18:42 [DEBUG] End NewClient
--- PASS: TestNewClient_WhenTLSSkipVerifyIsDisabled_ThenInsecureSkipVerifyIsFalse (0.00s)
=== RUN   TestNewClient_WhenClientIDIsEmpty_ThenThrowError
2023/08/08 17:18:42 [DEBUG] Init NewClient
--- PASS: TestNewClient_WhenClientIDIsEmpty_ThenThrowError (0.00s)
=== RUN   TestNewClient_WhenClientSecretIsEmpty_ThenThrowError
2023/08/08 17:18:42 [DEBUG] Init NewClient
--- PASS: TestNewClient_WhenClientSecretIsEmpty_ThenThrowError (0.00s)
=== RUN   TestNewClient_WhenControlPlaneIsEmpty_ThenThrowError
2023/08/08 17:18:42 [DEBUG] Init NewClient
--- PASS: TestNewClient_WhenControlPlaneIsEmpty_ThenThrowError (0.00s)
PASS
ok      github.com/cyralinc/terraform-provider-cyral/client     (cached)
=== RUN   TestAccDatalabelDataSource
=== PAUSE TestAccDatalabelDataSource
=== RUN   TestAccIntegrationIdPSAMLDataSource
=== PAUSE TestAccIntegrationIdPSAMLDataSource
=== RUN   TestAccLoggingIntegrationDataSource
=== PAUSE TestAccLoggingIntegrationDataSource
=== RUN   TestAccRepositoryDataSource
=== PAUSE TestAccRepositoryDataSource
=== RUN   TestAccRoleDataSource
=== PAUSE TestAccRoleDataSource
=== RUN   TestAccSAMLCertificateDataSource
=== PAUSE TestAccSAMLCertificateDataSource
=== RUN   TestAccSAMLConfigurationDataSource
=== PAUSE TestAccSAMLConfigurationDataSource
=== RUN   TestAccSidecarBoundPortsDataSource
=== PAUSE TestAccSidecarBoundPortsDataSource
=== RUN   TestAccSidecarCftTemplateDataSource
=== PAUSE TestAccSidecarCftTemplateDataSource
=== RUN   TestAccSidecarIDDataSource
=== PAUSE TestAccSidecarIDDataSource
=== RUN   TestAccSidecarInstanceIDsDataSource
=== PAUSE TestAccSidecarInstanceIDsDataSource
=== RUN   TestAccSidecarListenerDataSource
=== PAUSE TestAccSidecarListenerDataSource
=== RUN   TestIntegrationsData_GetValue_Default
--- PASS: TestIntegrationsData_GetValue_Default (0.00s)
=== RUN   TestIntegrationsData_GetValue_Splunk
--- PASS: TestIntegrationsData_GetValue_Splunk (0.00s)
=== RUN   TestAccProvider
2023/08/10 20:24:32 [DEBUG] Init dataSourceSidecarListener
2023/08/10 20:24:32 [DEBUG] End dataSourceSidecarListener
--- PASS: TestAccProvider (0.01s)
=== RUN   TestAccDatalabelResource
=== PAUSE TestAccDatalabelResource
=== RUN   TestAccDatadogIntegrationResource
=== PAUSE TestAccDatadogIntegrationResource
=== RUN   TestAccELKIntegrationResource
=== PAUSE TestAccELKIntegrationResource
=== RUN   TestAccHCVaultIntegrationResource
=== PAUSE TestAccHCVaultIntegrationResource
=== RUN   TestAccIntegrationIdPSAMLDraftResource
=== PAUSE TestAccIntegrationIdPSAMLDraftResource
=== RUN   TestAccIntegrationIdPSAMLResource
=== PAUSE TestAccIntegrationIdPSAMLResource
=== RUN   TestAccIdPIntegrationResource
=== PAUSE TestAccIdPIntegrationResource
=== RUN   TestAccLogsIntegrationResourceCloudWatch
=== PAUSE TestAccLogsIntegrationResourceCloudWatch
=== RUN   TestAccLogsIntegrationResourceDataDog
=== PAUSE TestAccLogsIntegrationResourceDataDog
=== RUN   TestAccLogsIntegrationResourceElk
=== PAUSE TestAccLogsIntegrationResourceElk
=== RUN   TestAccLogsIntegrationResourceElkEmptyEsCredentials
=== PAUSE TestAccLogsIntegrationResourceElkEmptyEsCredentials
=== RUN   TestAccLogsIntegrationResourceSplunk
=== PAUSE TestAccLogsIntegrationResourceSplunk
=== RUN   TestAccLogsIntegrationResourceSumologic
=== PAUSE TestAccLogsIntegrationResourceSumologic
=== RUN   TestAccLogsIntegrationResourceFluentbit
=== PAUSE TestAccLogsIntegrationResourceFluentbit
=== RUN   TestAccLogstashIntegrationResource
=== PAUSE TestAccLogstashIntegrationResource
=== RUN   TestAccLookerIntegrationResource
=== PAUSE TestAccLookerIntegrationResource
=== RUN   TestAccDuoMFAIntegrationResource
=== PAUSE TestAccDuoMFAIntegrationResource
=== RUN   TestAccPagerDutyIntegrationResource
=== PAUSE TestAccPagerDutyIntegrationResource
=== RUN   TestAccSlackAlertsIntegrationResource
=== PAUSE TestAccSlackAlertsIntegrationResource
=== RUN   TestAccSplunkIntegrationResource
=== PAUSE TestAccSplunkIntegrationResource
=== RUN   TestAccSumoLogicIntegrationResource
=== PAUSE TestAccSumoLogicIntegrationResource
=== RUN   TestAccMsTeamsIntegrationResource
=== PAUSE TestAccMsTeamsIntegrationResource
=== RUN   TestAccPolicyRuleResource
=== PAUSE TestAccPolicyRuleResource
=== RUN   TestPolicyRuleResourceUpgradeV0
--- PASS: TestPolicyRuleResourceUpgradeV0 (0.00s)
=== RUN   TestAccPolicyResource
=== PAUSE TestAccPolicyResource
=== RUN   TestAccRepositoryAccessGatewayResource
=== PAUSE TestAccRepositoryAccessGatewayResource
=== RUN   TestAccRepositoryAccessRulesResource
=== PAUSE TestAccRepositoryAccessRulesResource
=== RUN   TestAccRepositoryBindingResource
=== PAUSE TestAccRepositoryBindingResource
=== RUN   TestAccRepositoryConfAnalysisResource
=== PAUSE TestAccRepositoryConfAnalysisResource
=== RUN   TestRepositoryConfAnalysisResourceUpgradeV0
--- PASS: TestRepositoryConfAnalysisResourceUpgradeV0 (0.00s)
=== RUN   TestAccRepositoryConfAuthResource
=== PAUSE TestAccRepositoryConfAuthResource
=== RUN   TestRepositoryConfAuthResourceUpgradeV0
--- PASS: TestRepositoryConfAuthResourceUpgradeV0 (0.00s)
=== RUN   TestAccRepositoryDatamapResource
=== PAUSE TestAccRepositoryDatamapResource
=== RUN   TestAccRepositoryNetworkAccessPolicyResource
=== PAUSE TestAccRepositoryNetworkAccessPolicyResource
=== RUN   TestAccRepositoryResource
=== PAUSE TestAccRepositoryResource
=== RUN   TestAccRepositoryUserAccountResource
=== PAUSE TestAccRepositoryUserAccountResource
=== RUN   TestAccRoleSSOGroupsResource
=== PAUSE TestAccRoleSSOGroupsResource
=== RUN   TestRoleSSOGroupsResourceUpgradeV0
--- PASS: TestRoleSSOGroupsResourceUpgradeV0 (0.00s)
=== RUN   TestAccRoleResource
=== PAUSE TestAccRoleResource
=== RUN   TestAccSidecarCredentialsResource
=== PAUSE TestAccSidecarCredentialsResource
=== RUN   TestSidecarListenerResource
=== PAUSE TestSidecarListenerResource
=== RUN   TestAccSidecarResource
=== PAUSE TestAccSidecarResource
=== RUN   TestElementsMatch
--- PASS: TestElementsMatch (0.00s)
=== CONT  TestAccDatalabelDataSource
=== CONT  TestSidecarListenerResource
=== CONT  TestAccDatalabelResource
=== CONT  TestAccLogsIntegrationResourceFluentbit
=== CONT  TestAccLogstashIntegrationResource
=== CONT  TestAccLogsIntegrationResourceSumologic
=== CONT  TestAccSAMLConfigurationDataSource
=== CONT  TestAccSidecarResource
=== CONT  TestAccLogsIntegrationResourceElkEmptyEsCredentials
--- PASS: TestAccLogsIntegrationResourceFluentbit (6.63s)
=== CONT  TestAccLogsIntegrationResourceSplunk
--- PASS: TestAccLogsIntegrationResourceSumologic (7.73s)
--- PASS: TestAccDatalabelResource (7.98s)
=== CONT  TestAccLogsIntegrationResourceDataDog
=== CONT  TestAccLogsIntegrationResourceElk
--- PASS: TestAccSAMLConfigurationDataSource (8.80s)
--- PASS: TestAccLogstashIntegrationResource (12.74s)
=== CONT  TestAccLogsIntegrationResourceCloudWatch
--- PASS: TestAccLogsIntegrationResourceElkEmptyEsCredentials (7.10s)
=== CONT  TestAccIntegrationIdPSAMLResource
--- PASS: TestAccLogsIntegrationResourceDataDog (5.90s)
=== CONT  TestAccIdPIntegrationResource
=== CONT  TestAccIntegrationIdPSAMLDraftResource
--- PASS: TestAccLogsIntegrationResourceSplunk (6.99s)
--- PASS: TestAccLogsIntegrationResourceElk (7.59s)
=== CONT  TestAccELKIntegrationResource
--- PASS: TestAccDatalabelDataSource (20.45s)
=== CONT  TestAccHCVaultIntegrationResource
=== CONT  TestAccRepositoryAccessGatewayResource
--- PASS: TestAccLogsIntegrationResourceCloudWatch (8.30s)
=== CONT  TestAccSidecarCredentialsResource
--- PASS: TestAccELKIntegrationResource (6.96s)
--- PASS: TestAccIntegrationIdPSAMLDraftResource (10.95s)
=== CONT  TestAccRoleResource
=== CONT  TestAccRoleSSOGroupsResource
--- PASS: TestAccHCVaultIntegrationResource (7.43s)
=== CONT  TestAccRepositoryUserAccountResource
--- PASS: TestAccSidecarResource (29.28s)
=== CONT  TestAccRepositoryResource
--- PASS: TestAccSidecarCredentialsResource (7.06s)
--- PASS: TestSidecarListenerResource (34.89s)
=== CONT  TestAccDatadogIntegrationResource
--- PASS: TestAccIntegrationIdPSAMLResource (23.15s)
=== CONT  TestAccRepositoryConfAuthResource
=== CONT  TestAccSidecarInstanceIDsDataSource
--- PASS: TestAccRepositoryAccessGatewayResource (20.68s)
=== CONT  TestAccRepositoryBindingResource
--- PASS: TestAccDatadogIntegrationResource (8.41s)
--- PASS: TestAccRoleSSOGroupsResource (15.53s)
=== CONT  TestAccRepositoryConfAnalysisResource
=== CONT  TestAccSidecarListenerDataSource
--- PASS: TestAccRoleResource (22.40s)
=== CONT  TestAccSidecarIDDataSource
--- PASS: TestAccSidecarInstanceIDsDataSource (8.03s)
--- PASS: TestAccRepositoryConfAnalysisResource (10.15s)
=== CONT  TestAccSidecarBoundPortsDataSource
=== CONT  TestAccSidecarCftTemplateDataSource
--- PASS: TestAccRepositoryResource (23.34s)
--- PASS: TestAccRepositoryBindingResource (11.68s)
=== CONT  TestAccRepositoryNetworkAccessPolicyResource
=== CONT  TestAccDuoMFAIntegrationResource
--- PASS: TestAccRepositoryConfAuthResource (18.49s)
=== CONT  TestAccSlackAlertsIntegrationResource
--- PASS: TestAccSidecarIDDataSource (6.97s)
--- PASS: TestAccIdPIntegrationResource (43.08s)
=== CONT  TestAccRoleDataSource
--- PASS: TestAccRoleDataSource (6.99s)
=== CONT  TestAccLookerIntegrationResource
--- PASS: TestAccSlackAlertsIntegrationResource (8.21s)
=== CONT  TestAccPolicyRuleResource
=== CONT  TestAccPolicyResource
--- PASS: TestAccDuoMFAIntegrationResource (9.58s)
--- PASS: TestAccSidecarCftTemplateDataSource (12.86s)
=== CONT  TestAccMsTeamsIntegrationResource
=== CONT  TestAccRepositoryDatamapResource
--- PASS: TestAccSidecarListenerDataSource (20.55s)
--- PASS: TestAccSidecarBoundPortsDataSource (15.67s)
=== CONT  TestAccPagerDutyIntegrationResource
=== CONT  TestAccLoggingIntegrationDataSource
--- PASS: TestAccLookerIntegrationResource (8.94s)
=== CONT  TestAccSumoLogicIntegrationResource
--- PASS: TestAccPolicyResource (9.86s)
--- PASS: TestAccMsTeamsIntegrationResource (8.79s)
=== CONT  TestAccSAMLCertificateDataSource
--- PASS: TestAccPagerDutyIntegrationResource (9.34s)
=== CONT  TestAccIntegrationIdPSAMLDataSource
=== CONT  TestAccSplunkIntegrationResource
--- PASS: TestAccSAMLCertificateDataSource (4.80s)
--- PASS: TestAccRepositoryNetworkAccessPolicyResource (27.01s)
=== CONT  TestAccRepositoryDataSource
--- PASS: TestAccSumoLogicIntegrationResource (8.81s)
=== CONT  TestAccRepositoryAccessRulesResource
--- PASS: TestAccRepositoryUserAccountResource (55.02s)
--- PASS: TestAccLoggingIntegrationDataSource (12.94s)
--- PASS: TestAccPolicyRuleResource (23.44s)
--- PASS: TestAccSplunkIntegrationResource (8.86s)
--- PASS: TestAccRepositoryDatamapResource (26.55s)
--- PASS: TestAccIntegrationIdPSAMLDataSource (17.16s)
--- PASS: TestAccRepositoryDataSource (15.40s)
--- PASS: TestAccRepositoryAccessRulesResource (15.21s)
PASS
ok      github.com/cyralinc/terraform-provider-cyral/cyral      99.524s
```

### CODE REVIEW

File: cyral/resource_cyral_datalabel.go [hunk 1]
```diff
@@ -53,12 +40,44 @@ func resourceDatalabel() *schema.Resource {
...
+			"classification_rule": {
+				Description: "Classification rules are used by the " +
+					"[Automatic Data Map](https://cyral.com/docs/policy/automatic-datamap) feature to automatically map " +
+					"data locations to labels.",
+				Optional: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rule_type": {
```
Comment (wcmjunior) [inline cyral/resource_cyral_datalabel.go:52]: Please add a validation function.

File: cyral/resource_cyral_datalabel.go [hunk 2]
```diff
@@ -53,12 +40,44 @@ func resourceDatalabel() *schema.Resource {
...
+							Optional: true,
+							Default:  "UNKNOWN",
+						},
+						"rule_code": {
+							Description: "Actual code of the classification rule. For example, this attribute may contain " +
+								"REGO code for `REGO`-type classification rules.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"rule_status": {
```
Comment (wcmjunior) [inline cyral/resource_cyral_datalabel.go:65]: Please add a validation function.

**Review by wcmjunior — CHANGES_REQUESTED** [review]

File: cyral/resource_cyral_datalabel.go (hunk 2 above)
Comment (VictorGFM) [reply cyral/resource_cyral_datalabel.go:67]: Addressed: [Add validation to classification rule fields](https://github.com/cyralinc/terraform-provider-cyral/pull/436/commits/e029fe765aa9dae8aa866f4ede3b8e3b32bf2402)

File: cyral/resource_cyral_datalabel.go (hunk 1 above)
Comment (VictorGFM) [reply cyral/resource_cyral_datalabel.go:53]: Addressed: [Add validation to classification rule fields](https://github.com/cyralinc/terraform-provider-cyral/pull/436/commits/e029fe765aa9dae8aa866f4ede3b8e3b32bf2402)

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**